specs/Standards/Tokens/FungibleTokenMetadata.md

```diff
@@ -67,33 +52,125 @@ type FungibleTokenMetadata = {
     spec: string;
     name: string;
     symbol: string;
-    icon: string|null;
-    reference: string|null;
-    reference_hash: string|null;
     decimals: number;
+    homepage: string;
+    defaultIcon: bytes|null;
+    darkIcon: bytes|null;
+    extendedMetadata: dict;
 }
 ```
 
-**An implementing contract MUST include the following fields on-chain**
+Extended metadata key examples (from EtherScan, other token registries):
+
+* Creation date
+* Support email
+* Facebook
+* Twitter
+* Discord
+* Telegram
+* Reddit
+* LinkedIn
+* Company name 
+* Company address
+* Tags: a list of string tags that allow easier search of certain synthetically created tokens from indexers (Synthetic, Uniswap, others) - any service dynamically creating tokens my define these and they do not need to be displayed to a user
+
+Then CoinGecko like data
+
+* Orderbooks and AMMs
+   * Web browser viewable link of order books
+   * Machine readable link of trade data (price, vol 24h)
+
+# Discussion
+
+## On-chain vs. offchain hosted data
+
+Currently deployment on a NEAR token smart contract is 200 kb. The cost of 5 kb metadata on the cost of code deployment on the top of the contract deployment is miniscule. Icons are in the range of 5 kBytes - 200 kBytes per icon, so they are comparable to a smart contract deployment. 
+
+Off-chain data comes complicated with privacy and tracking issues as pointed out by Robert Zarembe.
+
+NEAR is the best blockchain in the world. For the reducing complexity, and after discussing with people on Discord, my stance is that everything on-chain. NEAR should scale for this as it scales for many things.
+
+## Name and symbol display limitations
+
+Any wallet and dApp MUST diplay a name up to 32 characters:
+
+WWWWWW WWWWWW WWWWWW WWWWWW 
+
+Use case: Synthetically created tokens: For example, a Uniswap-like pool can generate a token name and symbol:
+
+```
+Uniswap LP token for ETH-USDT pair
+LP ETH-USDT
+```
+
+## Icon format
+
+### Light theme vs. dark theme
+
+Two different icons needed: `defaultIcon` (black text on white background) for iOS, websites. `darkIcon` or websites supporting dark theme and Android UI. After discussing with Ethereum wallet vendors, there are multiple cases where icon looks good on one platform, but horrible on another, because of color bleed issues on the stroke lines of an icon. Token projects are not happy if their icon looks like amaeur.
+
+Thus, we should provide both light and dark icons by default.
+
+### SVG vs. PNG
+
+SVG is a Turing complete language with a [massive attack surface](https://www.fortinet.com/blog/threat-research/scalable-vector-graphics-attack-surface-anatomy). It may even contain embedded JavaScript. After discussing with infosec experts, I recommend SVG is not supported for icons because of decoding complexity and denial of service issues.
```

Comment:
This link greatly overstates the JS-in-SVG attack vector https://security.stackexchange.com/q/212894

Need to finish reading it later today, but wanted to at least share that link for now.